core/src/test/resources/applicationContext-dao.xml

```diff
@@ -17,6 +17,8 @@
 		     <list>
 			     <value>file:///#{systemEnvironment['PORTAL_HOME']}/src/main/resources/portal.properties</value>
                  <value>file:///#{systemEnvironment['PORTAL_HOME']}/src/main/resources/maven.properties</value>
+                 <value>file:///${PORTAL_HOME}/portal.properties</value>
+                 <value>classpath:portal.properties</value>
```

Comment:
Are you sure that this is correct ordering, and that that syntax to evaluate `$PORTAL_HOME` actually works in this context? This has gone wrong several times across the code base over time, leading to configurations varying from one XML to another.

Reply:
I feel like `$PORTAL_HOME/portal.properties` should _only_ be evaluated at runtime (never compile time), and `<portal_source>/src/main/resources/*.portal.properties` (which is the same thing as `classpath:portal.properties`) should be included at build time if at all, and this code seems to do the exact opposite.

Reply:
In addition, I feel like the runtime config file should override the buildtime defaults if it is in effect, and not the other way around.

Reply:
Spring documentation I bellieve to be relevant: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/io/support/PropertiesLoaderSupport.html#setLocations-org.springframework.core.io.Resource...-

Reply:
Even after moving Parquet Writer, the dao tests were failing without this change. Please see https://travis-ci.org/cBioPortal/cbioportal/jobs/553352211 which was a build after moving Parquet Writer.

Reply:
@doori Yes, I see.  TestDaoCancerStudy is looking for ${spark.app.name} which has to come from portal.properties .  I think you should follow the order specified in the business context file which will allow a portal.properties file that resides inthe PORTAL_HOME environment variable to override properties found in a "baked-in" portal.properties file:

[https://github.com/cBioPortal/cbioportal/blob/499beab3ca2b15e5ec525dfc2f3f7fe974d0ea31/business/src/main/resources/applicationContext-business.xml#L19](https://github.com/cBioPortal/cbioportal/blob/499beab3ca2b15e5ec525dfc2f3f7fe974d0ea31/business/src/main/resources/applicationContext-business.xml#L19)

Reply:
Thanks for the explanation. I moved the classpath: line before the systemEnvironment one.